pkg/image/image.go

```diff
@@ -392,26 +391,38 @@ func ChangeImageDomain(imageString string, newDomain string) (types.ImageReferen
 	return newImage, nil
 }
 
-func WithoutTag(image string) (string, error) {
-	fullImage, err := AddDefaultRegistry(image, "place.holder.com")
-	if err != nil {
-		return image, err
+// SplitImage Returns an image tag if it exists, an image digest if it exists, the image without the tag and digest, and any errors; in this order.
+// images with both a tag and digest are not supported
+func SplitImage(imageArg string) (string, string, string, error) {
```

Comment:
better if you return a struct (defined here)
type ImageParts struct {
    BaseImage string
    Tag string
    Digest string
}

Reply:
done